tests/onnx/quantization/test_ignore_post_processing_nodes.py

```diff
@@ -0,0 +1,205 @@
+                      'TopK_1': ['Identity_2'],
+                      'Identity_2': ['NMS_1', 'Conv_2'], 'NMS_1': ['Identity_3'], 'Identity_3': ['Output_1'],
+                      'Conv_2': ['Output_2']}
+        super().__init__(node_names, node_op_types, node_edges)
+        self.reference_ignored_scopes = ['Identity_3']
+
+
+class ModelToTest4(NNCFGraphToTest):
+    #          Input_1
+    #             |
+    #           Conv_1
+    #             |
+    #           Identity_1
+    #             |       \
+    #            TopK_1   Identity_4
+    #             |      /       |
+    #           Identity_2    Identity_5
+    #                \       /
+    #                 \     /
+    #                  NMS_1
+    #                    |
+    #                 Identity_3
+    #                    |
+    #                  Output_1
+
+    def __init__(self):
+        node_names = ['Input_1', 'Conv_1', 'Identity_1', 'TopK_1', 'Identity_2', 'NMS_1', 'Identity_3', 'Output_1',
+                      'Identity_4', 'Identity_5']
+        node_op_types = ['Input', 'Conv', 'Identity', 'TopK', 'Identity', 'NonMaxSuppression', 'Identity', 'Output',
+                         'Identity',
+                         'Identity']
+        node_edges = {'Input_1': ['Conv_1'], 'Conv_1': ['Identity_1'], 'Identity_1': ['TopK_1', 'Identity_4'],
+                      'TopK_1': ['Identity_2'],
+                      'Identity_2': ['NMS_1'], 'NMS_1': ['Identity_3'], 'Identity_3': ['Output_1'],
+                      'Identity_4': ['Identity_2', 'Identity_5'], 'Identity_5': ['NMS_1']}
+        super().__init__(node_names, node_op_types, node_edges)
+        self.reference_ignored_scopes = ['Identity_3', 'Identity_2', 'Identity_5', 'Identity_4', 'Identity_1']
+
+
+# pylint:disable=protected-access
+
+@pytest.mark.parametrize('model_to_test', [ModelToTest1(), ModelToTest2(), ModelToTest3(), ModelToTest4()])
+def test_add_ignoring_nodes_after_last_weight_node(model_to_test):
+    nncf_graph = model_to_test.nncf_graph
+
+    ip_graph = InsertionPointGraph(nncf_graph)
+    pattern = ONNX_HW_FUSED_PATTERNS.get_full_pattern_graph()
+    ip_graph = ip_graph.get_ip_graph_with_merged_hw_optimized_operations(pattern)
+
+    weight_nodes = nncf_graph.get_nodes_by_metatypes(WEIGHT_LAYER_METATYPES)
+    quantizable_layer_nodes = [QuantizableWeightedLayerNode(weight_node, [QuantizerConfig()]) for weight_node in
+                               weight_nodes]
+
+    solver = QuantizerPropagationSolver(quantizable_layer_nodes=quantizable_layer_nodes,
+                                        post_processing_metatypes=[ONNXTopKMetatype,
+                                                                   ONNXNonMaxSuppressionMetatype])
+
+    quantization_proposal = solver.run_on_ip_graph(ip_graph)
+    ignored_node_names = [nncf_graph.get_node_by_key(ignored_node_key).node_name for ignored_node_key in
+                          quantization_proposal._quant_prop_graph.ignored_node_keys]
```

Comment:
You are accessing a protected attribute here, which should have led you to an understanding that you are doing something wrong.

Note that if you refactor your code as I suggested above, introducing a new `PostprocessingNodeLocator` that, for instance, takes a `QPSG` and returns a list of node keys to ignore, then you can test that object separately from the `Solver`, and also in the common code.

Reply:
Done